dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -6175,23 +6195,19 @@ void gp_walk(const Item* item, void* arg)
         {
           ReturnedColumn* operand = NULL;
 
-          if (ifp->arguments()[i]->type() == Item::REF_ITEM)
-          {
-            Item* op = *(((Item_ref*)ifp->arguments()[i])->ref);
-            operand = buildReturnedColumn(op, *gwip, gwip->fatalParseError);
-          }
-          else
-            operand = buildReturnedColumn(ifp->arguments()[i], *gwip, gwip->fatalParseError);
+          operand =
+              buildReturnedColumn(directRefOrRefLoop(ifp->arguments()[i]), *gwip, gwip->fatalParseError);
```

Comment:
Same here, I am not clear on the usage of `directRefOrRefLoop()` in the above two places.

Reply:
directRefOrRefLoop either returns the first real Item or original Item if it isn't a ref.